openwisp_firmware_upgrader/api/views.py

```diff
@@ -30,33 +31,32 @@ class ListViewPagination(pagination.PageNumberPagination):
     max_page_size = 100
 
 
-class ProtectedAPIMixin(object):
-    authentication_classes = [BearerAuthentication, SessionAuthentication]
-    permission_classes = [DjangoModelPermissions]
+class ProtectedAPIMixin(FilterByOrganizationManaged):
+    authentication_classes = (
+        BearerAuthentication,
+        SessionAuthentication,
+    )
+    permission_classes = (
+        IsOrganizationManager,
+        DjangoModelPermissions,
+    )
     throttle_scope = 'firmware_upgrader'
     pagination_class = ListViewPagination
 
-
-class OrgAPIMixin(ProtectedAPIMixin):
     def get_queryset(self):
-        queryset = self.queryset.all()
-        if not self.request.user.is_superuser:
-            filter_key = f'{self.organization_field}__in'
-            user_orgs = self.request.user.organizations_dict.keys()
-            organization_filter = {filter_key: user_orgs}
-            queryset = queryset.filter(**organization_filter)
-        org = self.request.query_params.get('organization', None)
+        qs = super().get_queryset()
+        org_filtered = self.request.query_params.get('organization', None)
```

Comment:
What about superusers, I don't see the code to give them access to everything!

Reply:
@atb00ker this portion of code is to filter the queryset based on `organization` params if provided in the request, the queyset objects will be taken care by `ProtectedAPIMixin`, and `FilterByOrganizationManaged`.

Reply:
Read the code for the old version of this function, it returns `self.queryset.all()` for superusers, that's what I am talking about. We don't want any filtering for superusers, correct? 
They have global access to anything and everything.
There are no checks for superusers.

Reply:
@atb00ker if I can understand it correctly this portion of code is to filter the already received queryset based on the organization slug value, by passing a query param to the endpoint. if not passed this will returned whatever it gets from the class `FilterByOrganizzationManaged`.

PS: This portion of code was written by @nemesisdesign I have commited on top of it, by fixing the mixin classes and asseting the number of queries.